src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -283,7 +286,18 @@ public function getSearchContext(GridField $gridField)
                     . " or implement a getDefaultSearchContext() method on $modelClass"
                 );
             }
-            $this->searchContext = $singleton->getDefaultSearchContext();
+
+            $list = $gridField->getList();
+            $searchContext = $singleton->getDefaultSearchContext();
+
+            // In case we are working with an ArrayList we need to conver the search context into a basic search context
+            // This is because the scaffolded filters are inteded to use ORM for data searching,
+            // they rely on DataList functionality which is not avaialble on an ArayList
+            if ($list instanceof ArrayList) {
```

Comment:
```suggestion
            if (!($list instanceof DataList)) {
```
Allows for other list implementations that don't use the database (e.g. `EagerLoadedList` but also project-specific implementations people may make)